File: workspace/ASTIC-lib/src/ro/info/asticlib/math/Math.java
package ro.info.asticlib.math;

public class Math {
	public static int getApxlog(int base,int x){
		int power = 0;
		while(java.lang.Math.pow(base, power)<x){
			power++;
		}
		return power-1;
	}
	
	public static double computeCosine(double [] a,double []b){
        if(a.length!=b.length){
            return 0 ;
        }
        double topSum = 0,leftBSum=0,rightBSum=0;
        for(int i=0;i<a.length;i++){
            topSum+=(a[i]*b[i]);
            leftBSum+=java.lang.Math.pow(a[i],2) ;
            rightBSum+=java.lang.Math.pow(b[i], 2);
        }
        return topSum / (java.lang.Math.sqrt(leftBSum)*java.lang.Math.sqrt(rightBSum)) ;
    }
}

File: ASTIC_Client_VS2010Ex/ASTIC_Client_VS2010Ex/Math.cs
using System;

namespace ro.info.asticlib.math
{

	public class Math
	{
		public static int getApxlog(int @base, int x)
		{
			int power = 0;
			while (System.Math.Pow(@base, power) < x)
			{
				power++;
			}
			return power - 1;
		}

		public static double computeCosine(double[] a, double[] b)
		{
			if (a.Length != b.Length)
			{
				return 0;
			}
			double topSum = 0, leftBSum = 0, rightBSum = 0;
			for (int i = 0;i < a.Length;i++)
			{
				topSum += (a[i] * b[i]);
				leftBSum += System.Math.Pow(a[i],2);
				rightBSum += System.Math.Pow(b[i], 2);
			}
			return topSum / (System.Math.Sqrt(leftBSum) * System.Math.Sqrt(rightBSum));
		}
	}

}